### DISCUSSION TITLE
problem uploading firmware to esp 32

### DISCUSSION THREAD

**Discussion by azouzromdhane** [micropython/micropython#12028] [ESP32]
i have  a new esp32 board , i try to upload the firmware to it (bin file) but the Thonny IDE is showing the board is busy and i have to wait till the finish of the process or press ctr+c to exit ....so i m not able to upload the bin file , by the way the port is showing ok ...and it tryed the board on arduino IDE it works ok  ( see atteched picture plz)
regards
![shellThonny](https://github.com/micropython/micropython/assets/103148248/511daed8-73ce-4a5c-a055-9b6f032431a6)

**Comment by sosi-deadeye** [comment]
Please try first to flash your EP32 with `esptool.py`. I use `Thonny` frequently, but never used the upload method for the firmware.

**Reply by azouzromdhane** [reply]
thanks for reply 
can you give more details  plz...how to flash with esptool.py since i have no access to the borad

**Reply by 2dof** [reply]
tutorial [flashing-micropython-firmware-esptool-py](https://randomnerdtutorials.com/flashing-micropython-firmware-esptool-py-esp32-esp8266/).


or using  [upycraft ide](https://randomnerdtutorials.com/flash-upload-micropython-firmware-esp32-esp8266/)

**Reply by bixb922** [reply]
esptool.py works through the usb connector of the Esp32 board. See https://docs.espressif.com/projects/esptool/en/latest/esp32/#esptool-py-documentation how to install. See here https://micropython.org/download/esp32/ for the esptool command line needed to upload a Micropython image. The port name depends on the operating system of the PC (Mac, Windows, Linux). You will need the correct port name on the PC (for example COM5 or /dev/ttyUSB0). The mpremote utility can search for it (and do a lot more interesting things). On some boards you need to push the boot button before powering on, that changes the port name again.  This tutorial could also help https://docs.micropython.org/en/latest/esp32/tutorial/intro.html
